challenger/README.md

```diff
@@ -0,0 +1,124 @@
+
+1. **Task Manager ABI**: Get the ABI of the task manager from your bindings.
+
+   ```go
+      taskManagerAbi, err := cstaskmanager.ContractIncredibleSquaringTaskManagerMetaData.GetAbi()
+   ```
+
+2. **Challenger Configuration**: Create a `challenger.Config` struct with the following fields:
+   - `EthWsUrl`: The URL of the Ethereum websocket
+   - `Logger`: The logger
+   - `TaskManagerAbi`: The ABI of the task manager
+   - `EthClient`: The Ethereum client
+
+      ```go
+        ethHttpUrl := "http://localhost:8545"
+        ethHttpClient, _ := ethclient.Dial(ethHttpUrl)
+      
+        cfg := challenger.Config{
+          EthWsUrl:       "ws://localhost:8545",
+          Logger:         logger,
+          TaskManagerAbi: taskManagerAbi,
+          EthClient:      ethHttpClient,
+        }
+      ```
+
+3. **Task Manager**: Create a `TaskManager` to interact with the user defined task manager contract.
+   - This will be in charge of raising challenges.
+
+      ```go
+        ecdsaPrivateKey, _ := crypto.HexToECDSA(testutils.ANVIL_FIRST_PRIVATE_KEY)
+        txMgr, _ := txmgr.NewSimpleTxManagerFromPrivateKey(logger, ethHttpClient, ecdsaPrivateKey)
+  
+        challengerRaiser, err := taskmanager.NewTaskManagerFromAbi[*big.Int, *big.Int](
+          taskManagerAddress,
+          taskManagerAbi,
+          txMgr,
+          ethHttpClient,
+        )
+      ```
+
+4. **Task Verification Logic**: Define a function that computes the expected result for a task, which will be used to verify operator responses
+   - This would be the logic to compute a new task.
+
+      ```go
+        func square(taskIndex uint32, numberToSquare *big.Int) (*big.Int, error) {
+          numberSquared := big.NewInt(0).Exp(numberToSquare, big.NewInt(2), nil)
+          return numberSquared, nil
+        }
+      ```
+
+5. **Response Calculator**: To abstract your computation into the challenger, we provide a `ResponseCalculator` interface with a standar `functionResponseCalculator` struct. This struct implements the interface and a helper method for turning your function into `functionResponseCalculator`:
+   - `NewFunctionResponseCalculator`: Create a response calculator from your computation function.
+
+      ```go
+        calculator := operator.NewFunctionResponseCalculator(square)
+      ```
+
+6. **Verifier**: Create a verifier from the response calculator.
+   - This will be in charge of computing the response of a task and will use a user-defined function to compare the computed response with the operator's response.
+
+      ```go
+        // BigIntEqual receives two *big.Int and returns whether they are equal
+        func BigIntEqual(a, b *big.Int) bool {
+          return a.Cmp(b) == 0
+        }
+
+        validation := challengerprocessor.ResponseValidationFunctionFromResponseCalculator(calculator, BigIntEqual)
+      ```
+
+7. **Challenger Task Processor**: Create a [`ChallengerTaskProcessor`] interface implementation.
```

Comment:
We should add a section specifying how one would go about for creating a ChallengerTaskProcessor implementation. Same for the other modules.